### ISSUE TITLE
trying to run "sendtoaddress" RPC in Bitcoind

### ISSUE THREAD

**Issue by pepper28** [JSKitty/bitcoin-rpc#1]
Could you please provide an example of how you would make the call to Bitcoin Core's "sendtoaddress" method.  I will also need to preset the "walletpassphrase" password phrase prior to calling sendtoaddress.  Thank you in advance for your help.

**Comment by JSKitty** [comment]
Hey @pepper28! No problem, the library has a very 'literal' approach to Bitcoin RPC parameters, so for example, with the [latest RPC docs here](https://bitcoincore.org/en/doc/22.0.0/rpc/wallet/sendtoaddress/).

A simple unlock + transfer would look like:
```js
const RPCClient = require('@jskitty/bitcoin-rpc');

// Create a new RPC class for a Bitcoind RPC daemon
//                        user    pass    host         port
const btc = new RPCClient('user', 'pass', '127.0.0.1', 8332);

// Unlock the wallet for 15 seconds (afterwards, it will auto-lock)
btc.call("walletpassphrase", "password123", 15).then(() => {

    // Create a transfer of 0.1 BTC to an address
    btc.call("sendtoaddress", "1DGNa15ZQX0xUgFoqJ2i7Z2DPU2J6hw62i", 0.1).then(res => {
        console.log("TX Hash: " + res); // This should be a TX hash!
    });

});
```

(this is untested code, but *should* work, just do some obvious adjustments such as: RPC information, password, bitcoin address and amount)

**Comment by pepper28** [comment]
Thank you so MUCH!!!!

**Comment by JSKitty** [comment]
Hope all went well! @pepper28 closing this issue for now, I assume you've gotten what you need. :pray:

**Comment by pepper28** [comment]
Yes thank you very much. Your solution works like a charm.

Sent from my iPhone

On Nov 3, 2022, at 5:23 PM, JSKitty ***@***.***> wrote:

﻿

Closed #1<https://github.com/JSKitty/bitcoin-rpc/issues/1> as completed.
